### ISSUE TITLE
[sklearn#34268] Add an option to use Latin Hypercube Sampling to `RandomizedSearchCV` and `HalvingSearchCV`

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#341]
**Mirrored from [scikit-learn#34268](https://github.com/scikit-learn/scikit-learn/issues/34268)**

Original title: Add an option to use Latin Hypercube Sampling to `RandomizedSearchCV` and `HalvingSearchCV`
Original labels: New Feature, Needs work, Needs Decision - Include Feature
Created: 2026-06-12T13:21:51Z

---

> [!WARNING]
> This issue is not yet ready for a PR. If you are interested in contributing to scikit-learn, please have a look at our [contributing guidelines](https://scikit-learn.org/dev/developers/contributing.html), and in particular the sections for [new contributors](https://scikit-learn.org/dev/developers/contributing.html#new-contributors) and the ["Needs triage"](https://scikit-learn.org/dev/developers/contributing.html#issues-tagged-needs-triage) label.

### Describe the workflow you want to enable

[Latin Hypercube Sampling ](https://en.wikipedia.org/wiki/Latin_hypercube_sampling) is an alternative to uniform sampling that prevent clumping sampled values.

It was empirically shown to reduce the variance and improve the average result of randomized search by better covering the search space, in particular when running RS with a limited number of iterations on a medium to large dimensional parameters search space:

https://www.r-bloggers.com/2019/07/latin-hypercube-sampling-in-hyper-parameter-optimization/

### Describe your proposed solution

I think we could introduce a new `sampling_strategy = {"latin_hypercube", "uniform"}` parameter to the `RandomizedSearchCV` and `HalvingSearchCV` classes.

Implementing LHC sampling should be trivial for discretized search spaces and the extra code complexity should be very limited. For parameter ranges specified as `scipy.stats` distribution objects, it should be possible to map LHC samples from the 0-1 range to the actual param space using the inverse CDF functions.

### Additional context

Arguably, the linked blog post is not a proper peer reviewed paper, but I doubt that such a simple yet practically seemingly incremental improvement over RS has a chance of ever being published in a peer reviewed journal/conference and get a large enough citation count to meet our formal inclusion criteria.

Implementing a prototype and empirically evaluating it on a large and diverse enough set of HP tuning problems (varying hyper-parameter dimensionality, number of candidates, kinds of data and pipelines to ensure diversity of sensitivity of perf metric to hparam values...) could help us decide whether this option would be worthy of inclusion or not. Interested contributors, feel free to create a dedicated prototype notebook or repo and share the results as a comment to this issue.

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to https://github.com/godofecht/flow-scikit/issues/82 in the flow-scikit repository. Further library-specific triage and implementation will happen there.
